test/runtime/memcache.go

```diff
@@ -106,99 +109,200 @@ var _ = Describe("RuntimeMemcacheBinary", func() {
 	})
 
 	AfterFailed(func() {
+		containers("delete")
 		vm.ReportFailed("cilium policy get")
 	})
 
-	It("Tests basic memcache operation", func() {
-		key := "test1"
-		value := "value1"
-		By("Setting memcache value")
-		r := setKey(key, value)
-
-		r.ExpectSuccess("Unable to set key", key)
-
-		By("getting memcache value")
-		r = getKey(key)
-		r.ExpectSuccess("Unable to get key", key)
-		r.ExpectContains(value, "Value mismatch on get")
-	})
-
-	It("Tests policy allowing all actions", func() {
-		_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-binary-allow.json"), 300)
-		Expect(err).Should(BeNil(), "Failed to import policy")
-
-		key := "test2"
-		value := "value2"
-		By("Setting memcache value")
-		r := setKey(key, value)
-		r.ExpectSuccess("Unable to set key", key)
-
-		By("getting memcache value")
-		r = getKey(key)
-		r.ExpectSuccess("Unable to get key", key)
-		r.ExpectContains(value)
-	})
-
-	It("Tests policy disallowing set action", func() {
-		By("Setting key before disallowing set")
-		keyBeforePolicy := "before"
-		valBeforePolicy := "beforeval"
-		r := setKey(keyBeforePolicy, valBeforePolicy)
-		r.ExpectSuccess("Unable to set key %q without policy loaded", keyBeforePolicy)
-
-		By("Importing policy disallowing set")
-		_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-binary-disallow-set.json"), 300)
-		Expect(err).Should(BeNil(), "Failed to import policy")
-
-		By("Trying to set new key")
-		key := "keyAfterPolicy"
-		value := "valAfterPolicy"
-
-		By("Setting memcache value")
-		r = setKey(key, value)
-		r.ExpectFail("Set key should be prohibited by policy")
-
-		By("getting memcache value")
-		r = getKey(keyBeforePolicy)
-		r.ExpectSuccess("Unable to get key set before applying policy")
-		r.ExpectContains(valBeforePolicy, "Value mismatch on get")
+	Context("Testing binary memcache", func() {
+		It("Tests basic memcache operation", func() {
+			key := "test1"
+			value := "value1"
+			By("Setting memcache value")
+			r := setKeyBinary(key, value)
+
+			r.ExpectSuccess("Unable to set key %s", key)
+
+			By("getting memcache value")
+			r = getKeyBinary(key)
+			r.ExpectSuccess("Unable to get key %s", key)
+			r.ExpectContains(value, "Value mismatch on get")
+		})
+
+		It("Tests policy allowing all actions", func() {
+			_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-allow.json"), 300)
+			Expect(err).Should(BeNil(), "Failed to import policy")
+
+			key := "test2"
+			value := "value2"
+			By("Setting memcache value")
+			r := setKeyBinary(key, value)
+			r.ExpectSuccess("Unable to set key %s", key)
+
+			By("getting memcache value")
+			r = getKeyBinary(key)
+			r.ExpectSuccess("Unable to get key %s", key)
+			r.ExpectContains(value)
```

Comment:
Nit: provide an error string as the second argument (value mismatch for get for key %s) or something similar.